base/compiler/abstractinterpretation.jl

```diff
@@ -388,7 +387,7 @@ function abstract_iteration(@nospecialize(itertype), vtypes::VarTable, sv::Infer
     valtype = statetype = Bottom
     ret = Any[]
     stateordonet = widenconst(stateordonet)
-    while !(Nothing <: stateordonet) && length(ret) < sv.params.MAX_TUPLETYPE_LEN
+    while !(Nothing <: stateordonet) || length(ret) < sv.params.MAX_TUPLE_SPLAT
```

Comment:
Why the change from `&&` to `||`? Doesn't that lead to an infinite loop for something like
```julia
struct Foo end
Base.iterate(::Foo) = (0, nothing)
Base.iterate(::Foo, state) = (0, (state,))
foo(x) = tuple(x...)
@code_warntype foo(Foo())
```
?

Reply:
Oops!!! :anguished: I'll fix that